lib/jxl/enc_heuristics.cc

```diff
@@ -984,6 +984,21 @@ Status ComputeARHeuristics(const FrameHeader& frame_header,
   const float c3 = std::pow(c3base, clamped_butteraugli);
   const float c4 = 1.247544678665836f;
   const float context_weight = c1 + c2 * clamped_butteraugli;
+  float mul[3 * 3 * 3] = {0};
+  for (uint8_t top_val : epf_steps) {
+    for (uint8_t left_val : epf_steps) {
+      for (uint8_t val : epf_steps) {
+        int context = epf_steps_lut[top_val] * 3 + epf_steps_lut[left_val];
+        const auto& ctx_histo = histo[context];
+        const int mulix = epf_steps_lut[val] + 3 * context;
+        mul[mulix] = 1.0 / (c4 + std::log1p(ctx_histo[val] * context_weight /
```

Comment:
Can `c4 + std::log1p(ctx_histo[val] * context_weight / totals[context])` ever be zero?

Reply:
no, c4 is positive, result of log1p is >= 0 because all remaining computation is positive or zero